pydantic/_internal/_model_construction.py

```diff
@@ -451,6 +455,7 @@ def complete_model_class(
         config_wrapper: The config wrapper instance.
         raise_errors: Whether to raise errors.
         types_namespace: Optional extra namespace to look for types in.
+        is_dynamic_model: Whether the model is a dynamic model (function is called from `create_model`).
```

Comment:
no need for this argument, and logic here. Instead put the logic in `create_model`